📖 (docs) draft catalogd docs with new metas endpoint (#1841)
operator-framework/operator-controller: docs/howto/catalog-queries.md

    By default, Catalogd is installed with TLS enabled for the catalog webserver.
    The following examples will show this default behavior, but for simplicity's sake will ignore TLS verification in the curl commands using the `-k` flag.

!!! note 

>     While using the `/api/v1/metas` endpoint shown in the below examples, it is important to note that the endpoint accepts parameters following the pattern `/api/v1/metas?<parameters>`. Valid parameter set is a set that corresponds to any combination of the fields of the [FBC Meta structure](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#schema) (except for `Blob`).

Reviewer: Same here, should "Structure" be capitalized?

>     While using the `/api/v1/metas` endpoint shown in the below examples, it is important to note that the endpoint accepts parameters following the pattern `/api/v1/metas?<parameters>`. Valid parameter set is a set that corresponds to any combination of the fields of the [FBC Meta structure](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#schema) (except for `Blob`).

Author: I think I avoided using that phrase because we'd named the endpoint `query` before switching over to `metas`. So this is definitely a case of me overthinking things 😅 I've edited this area a little bit, hope it's clearer now.

>     While using the `/api/v1/metas` endpoint shown in the below examples, it is important to note that the endpoint accepts parameters following the pattern `/api/v1/metas?<parameters>`. Valid parameter set is a set that corresponds to any combination of the fields of the [FBC Meta structure](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#schema) (except for `Blob`).

Reviewer 2: feel free to resolve - I can't do that with my permissions

>     eg. `schema=<schema_name>&package=<package_name>`, `schema=<schema_name>&name=<name>`, `package=<package_name>&name=<name>` are all valid paramet combination. However `schema=<schema_name>&version=<version_string>` is not a valid parameter combination, since version is not a first class FBC meta field. 

Reviewer: ```suggestion
    e.g. `schema=<schema_name>&package=<package_name>`, `schema=<schema_name>&name=<name>`, and `package=<package_name>&name=<name>` are all valid parameter combinations. However `schema=<schema_name>&version=<version_string>` is not a valid parameter combination, since version is not a first class FBC meta field. 
```

    By default, Catalogd is installed with TLS enabled for the catalog webserver.
    The following examples will show this default behavior, but for simplicity's sake will ignore TLS verification in the curl commands using the `-k` flag.

> !!! note 

Reviewer 2: this doesn't render correctly, I think the syntax is [!NOTE], or maybe [!IMPORTANT]?

Similar case for https://github.com/operator-framework/operator-controller/pull/1841/files#diff-6295b5bd58ccf02e628fe42adbc43f32b9df3f3ad4ab61ac813366162d65a490R54 and https://github.com/operator-framework/operator-controller/pull/1841/files#diff-944a5a8340da19a891934706c3a6a9f5eeded7eacdce3c107edd1f9dfdd7f173R36

Reviewer 2: resolved in https://github.com/operator-framework/operator-controller/pull/1841#discussion_r1995038039

* Package metadata:
    ``` terminal
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.package&name=<package_name>'
    ```

    `<package_name>`
    : Name of the package from the catalog you are querying.

* Blobs that belong to a package (that are not schema=olm.package):
    ``` terminal
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?package=<package_name>'
    ```

    `<package_name>`
    : Name of the package from the catalog you are querying.

Note: the `olm.package` schema blob does not have the `package` field set. In other words, to get all the blobs that belong to a package, along with the olm.package blob for that package, a combination of both of the above queries need to be used. 

## Channel queries

* Channels in a package:
    ``` terminal
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.channel&package=<package_name>'
    ```

    `<package_name>`
    : Name of the package from the catalog you are querying.

* Versions in a channel:
    ``` terminal

>     curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.channel&package=<package_name>' | jq -s '.[] | select( .name == "<channel_name>" ) | .entries | .[] | .name'

Reviewer 3: I would instead do
```suggestion
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.channel&package=<package_name>&name=<channel_name>' 
```

* Bundles in a package:
    ``` terminal
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.bundle&package=<package_name>'
    ```

    `<package_name>`
    : Name of the package from the catalog you are querying.

* Bundle dependencies and available APIs:
    ``` terminal

>     curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.bundle&name=<bundle_name>' | jq -s '.[] | .properties | .[] | select(.type=="olm.gvk")'

Reviewer 3: Yas! 🎆

>     curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.bundle&name=<bundle_name>' | jq -s '.[] | .properties | .[] | select(.type=="olm.gvk")'

Reviewer 3: (although this syntax could be simplified as ...)
```suggestion
    curl -k 'https://localhost:8443/catalogs/operatorhubio/api/v1/metas?schema=olm.bundle&name=<bundle_name>' | jq -s '.[] | .properties[] | select(.type=="olm.gvk")'
```
